src/middlewares/clientAuthMiddleware.js

```diff
@@ -0,0 +1,77 @@
+import client from 'Config/client'; // eslint-disable-line import/no-unresolved
+import { errorHttpResponseAfterHandler } from './errorHttpResponseMiddleware';
+import validateFields from '../utils/validateFields';
+import { LesgoException } from '../exceptions';
+
+const FILE = 'Middlewares/clientAuthMiddleware';
+
+const validateParams = params => {
+  const validFields = [
+    { key: 'x-client-id', type: 'string', required: true },
+    { key: 'client', type: 'object', required: true },
+  ];
+
+  try {
+    return validateFields(params, validFields);
+  } catch (error) {
+    throw new LesgoException(error.message, `${FILE}::INVALID_AUTH_DATA`, 403, {
+      error,
+    });
+  }
+};
+
+const getClientKey = event => {
+  if (typeof event.headers['x-client-id'] === 'string') {
+    return event.headers['x-client-id'];
+  }
+
+  if (event.input && typeof event.input.clientid === 'string') {
+    return event.input.clientid;
+  }
+
+  return undefined;
+};
+
+export const clientAuthMiddlewareBeforeHandler = (
+  handler,
+  next,
+  func = undefined
```

Comment:
@jpaylaga maybe let's make this more uniform with the rest of the middlewares, and put this under `opts`. So we can add more in the future without doing any breaking changes

Reply:
Thanks @xXLXx I have replaced the `func` parameter with `opt` with additional backwards compatibility as POCs might not work if not provided.